001-config-overhaul.md

```diff
@@ -648,6 +688,138 @@ hiccups along the way.
+   - Is there a valid use case for displaying file paths or other meta-configuration data?
+   - The initial motivation for retrieving the configuration was to inspect its state after making run-time changes via
+     the configuration update endpoint. However, this becomes redundant if such modifications are discouraged or
+     removed.
+
+#### Recommendation
+
+For the reasons stated above, it's recommended to remove the configuration-related endpoints from Iroha. This step will
+reduce potential points of failure, ambiguities, and complexities. Administrators should be encouraged to use the
+comprehensive configuration reference for any documentation needs and to restart Iroha with an updated configuration
+when changes are required.
+
+#### Summary
+
+This proposal advocates for the removal of configuration-related endpoints in Iroha, emphasising system simplicity,
+clarity, and robustness over mutable configurations during runtime. The configuration reference will serve as the sole
+authoritative documentation source, and Iroha restarts are recommended for configuration updates.
+
+### Proposal 10 - Standardise Relative Paths Resolution
+
+#### Objective
+
+To define a consistent and intuitive method for resolving relative file system paths in the Iroha configuration.
+
+#### Rationale
+
+Currently, relative paths in the configuration file are resolved based on the current working directory (CWD), which can
+lead to unpredictable results, specifically if Iroha is executed from a directory different from where the config file
+is located. To address this:
+
+1. **Config File Relative Paths**: Relative paths specified within the config file should be resolved relative to the
+   location of the config file itself, not the CWD.
+2. **Environment Variable Paths**: Paths set through environment variables corresponding to configuration parameters
+   should be resolved relative to the CWD.
+3. **Documentation**: This behaviour will be explicitly documented in the
+   [configuration reference](#proposal-2---reference-before-implementation) to ensure users are aware of how relative
+   paths are handled.
+
+Standardising how relative paths are resolved can help prevent potential issues and confusion for users.
+
+#### Summary
+
+This proposal seeks to bring consistency and clarity to how relative file paths are resolved in Iroha's configuration.
+Paths in the config file will be relative to its location, while paths from environment variables will be relative to
+the CWD.
+
+### Proposal 11 - Adopt Same Proposals for Iroha Client CLI
+
+#### Objective
+
+To apply the configuration improvement proposals devised for Iroha 2 to the `iroha_client_cli`.
+
+#### Rationale
+
+1. **Shared Foundation**: The iroha_client_cli and Iroha 2 are built on the same foundation, and the client CLI's
+   configuration structure is similar to Iroha's.
+2. **Consistent Experience**: Providing a consistent configuration experience between Iroha and its official client
```

Comment:
Should we address SDK's client configuration as well then 🤔

Reply:
Hmm, not sure about that. How much is its configuration and UI the same as `iroha_client_cli`?

Reply:
Speaking about `iroha_client_cli` it's something like a configured instance of Java SDK's [Iroha2Client](https://github.com/hyperledger/iroha-java/blob/007a9ac55991cd8a2b519e62a10144156d9f8301/modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt#L86) expanded with an account, key and it's possible just to send single instructions via the cli.

So it might be worth adding an SDK section rather to align configurations/structure among just SDKs.

Reply:
I see that configuration proposals might be useful for other Iroha clients too, not only the Rust CLI. It might be even useful for any other Iroha-related CLIs. However, I don't clearly see that there are other candidates to adopt most of the proposals. So, I would like to mention only `iroha_client_cli` in the RFC. It doesn't prevent us from adopting proposals to anything else in the future if we see a benefit from it.